File: 2025-10/2025-10-26/duration_formatter.py
def format(seconds):
    hrs = seconds // 3600
    mins = (seconds - hrs * 3600) // 60
    secs = seconds - hrs * 3600 - mins * 60

    duration = f"{hrs}:" if hrs != 0 else ""
    duration += f"0{mins}:" if hrs != 0 and mins <= 9 else f"{mins}:"
    duration += f"0{secs}" if secs <= 9 else f"{secs}"

    return duration

print(format(500))
print(format(4000))
print(format(1))
print(format(5555))
print(format(99999))

File: 2025-10/2025-10-26/durationFormatter.js
function format(seconds) {
  const hrs = Math.floor(seconds / 3600);
  const mins = Math.floor((seconds - hrs * 3600) / 60);
  const secs = seconds - hrs * 3600 - mins * 60;

  let duration = hrs != 0 ? `${hrs}:` : "";
  duration += hrs != 0 && mins <= 9 ? `0${mins}:` : `${mins}:`;
  duration += secs <= 9 ? `0${secs}` : `${secs}`;

  return duration;
}

console.log(format(500));
console.log(format(4000));
console.log(format(1));
console.log(format(5555));
console.log(format(99999));